### ISSUE TITLE
bundle-stats.html not generated if compare is true

### ISSUE THREAD

**Issue by ktw1016** [relative-ci/bundle-stats#1067]
I have a react app that uses bundle-stats. Currently updating webpack 5, I'm also updating bundle-stats. I removed `bundle-stats@1.10.1` (latest version working) and installed `bundle-stats-webpack-plugin@2.3.0`. I noticed `bundle-stats.html` not generating with when `compare` is `true`. However, it works as expected when `compare` is `false`.

Node.js 10.14.2
webpack: 5.9.0
bundle-stats-webpack-plugin: 2.3.0

Webpack config that might be relevant:
```
{
  context: '$home/app/client'
}
```

bundle stats plugin:
```
BundleStatsWebpackPlugin({
        compare: true,
        json: true,
        outDir: "..",
      })
```

**Comment by vio** [comment]
Thanks for raising the issue @ktw1016, will be helpful if possible to share more details about your issue:
1. are you getting any errors / warnings?
2. are the reports generated?

**Comment by ktw1016** [comment]
Found out that with config `baseline=true`, reports are not generated at all if there is already `node_modules/.cache/bundle-stats/baseline.json`. Here is the output:
```
**Read baseline from ../../../node_modules/.cache/bundle-stats/baseline.json
Write baseline data to ../../../node_modules/.cache/bundle-stats/baseline.json
Bundle Size — 32.6MB (0%).**
en:
  assets by chunk 22.5 MiB (id hint: vendors) 19 assets
  56 assets
  orphan modules 347 KiB [orphan] 165 modules
  runtime modules 6.5 KiB 12 modules
  modules by path ./node_modules/ 7.43 MiB 1788 modules
  modules by path ./src/ 2.37 MiB 569 modules
  ../data/faq.csv 12.5 KiB [built] [code generated]
  assert (ignored) 15 bytes [built] [code generated]
  en (webpack 5.9.0) compiled successfully in 43217 ms
```

However, if `baseline.json` is missing, report is generated. Output:
```
**Missing baseline stats, see "baseline" option.
Write baseline data to ../../../node_modules/.cache/bundle-stats/baseline.json
Bundle Size — 32.6MB (+100%).**
en:
  assets by status 32.6 MiB [compared for emit]
    assets by chunk 22.5 MiB (id hint: vendors) 19 assets
  assets by status 3.25 MiB [emitted]
    **asset ../bundle-stats.json 1.65 MiB [emitted]
    asset ../bundle-stats.html 1.17 MiB [emitted]**
    asset ../../../node_modules/.cache/bundle-stats/baseline.json 439 KiB [emitted]
  orphan modules 347 KiB [orphan] 165 modules
  runtime modules 6.5 KiB 12 modules
  modules by path ./node_modules/ 7.43 MiB 1788 modules
  modules by path ./src/ 2.37 MiB 569 modules
  ../data/faq.csv 12.5 KiB [built] [code generated]
  assert (ignored) 15 bytes [built] [code generated]
  en (webpack 5.9.0) compiled successfully in 42267 ms
```

**Comment by vio** [comment]
hi @ktw1016! I was able to replicate the issue on one of the example repos - https://github.com/relative-ci/example-bundle-stats-webpack-config/tree/e00452845d1cce478a2f183298623b057fd3792e. After I upgraded html-webpack plugin to fix a deprecation warning, the problem disappeared.

I'm not 100%, but I suspect a deeper problem with the new way Webpack 5 is processing assets. In the meantime I updated the way [bundle-stats is processing the reports](https://github.com/relative-ci/bundle-stats/pull/1105/commits/2bdde59d92bdb7533ce308b46aceedb3c1326bac) and released a beta version (`2.4.0-beta.6`).

If you manage to give it a try, let me know how is going.

**Comment by ktw1016** [comment]
Hi @vio , thanks for getting back to me. What's the html-webpack plugin you upgraded?

I tried with `bundle-stats-webpack-plugin@2.4.0-beta.6`, And I'm still getting the exact same error as described above. Slightly different output however.

Missing `node_modules/.cache/bundle-stats/baseline.json` output (report is generated):
Missing baseline stats, see "baseline" option.
Bundle Size — 4.7MB (+100%).
(node:126) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
	Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
	Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
en:
  assets by path *.js 4.7 MiB
    assets by chunk 2.76 MiB (id hint: vendors)
      assets by status 1.46 MiB [big]
        asset vendors-node_modules_graphiql_esm_index_js-node_modules_isomorphic-fetch_fetch-npm-browserify-359772.app-en.min.js 816 KiB [emitted] [minimized] [big] (id hint: vendors)
        asset vendors-node_modules_html2canvas_dist_html2canvas_js-node_modules_jspdf-autotable_dist_jspdf_-247707.app-en.min.js 682 KiB [emitted] [minimized] [big] (id hint: vendors) 1 related asset
      17 assets
  assets by path ../ 2.21 MiB
    asset ../************.json 1.19 MiB [emitted] [big]
    asset ../************.html 1.02 MiB [emitted] [big]
  orphan modules 2.46 MiB [orphan] 976 modules
  runtime modules 6.63 KiB 12 modules
  modules by path ./ 9.23 MiB
    modules by path ./node_modules/ 6.58 MiB 1155 modules
    modules by path ./src/ 2.65 MiB 352 modules
  ../data/faq.csv 12.5 KiB [built] [code generated]
  assert (ignored) 15 bytes [built] [code generated]

If there is already `node_modules/.cache/bundle-stats/baseline.json` (report not generated):
en:
  assets by chunk 21.3 MiB (id hint: vendors) 19 assets
  56 assets
  orphan modules 347 KiB [orphan] 165 modules
  runtime modules 6.62 KiB 12 modules
  modules by path ./node_modules/ 7.43 MiB 1788 modules
  modules by path ./src/ 2.37 MiB 569 modules
  ../data/faq.csv 12.5 KiB [built] [code generated]
  assert (ignored) 15 bytes [built] [code generated]
  en (webpack 5.10.2) compiled successfully in 48450 ms
Read baseline from ../../../node_modules/.cache/bundle-stats/baseline.json
Write baseline data to ../../../node_modules/.cache/bundle-stats/baseline.json
Bundle Size — 31.09MB (0%).

**Comment by vio** [comment]
@ktw1016 

> What's the html-webpack plugin you upgraded?
[html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin#webpack-5)

Thanks for trying `bundle-stats-webpack-plugin@2.4.0-beta.6`, but i was still missing couple bits. 

There is a new beta version available (`bundle-stats-webpack-plugin@2.4.0-beta.6`) that is solving the deprecation warning and is working for your project as well.

```shell
~/w/r/_/i/client on master ⨯ npm run IB_prod_stats                                                                                                                                                     00:20:03

> node build_code/webpack_starter.js PROD NO-WATCH STATS EN main_client

  app: main_client,
  prod: true,
  babel: true,
  languages: en,

Read baseline from ../../../node_modules/.cache/bundle-stats/baseline.json
Bundle Size — 4.68MB (0%).
en:
  assets by path *.js 4.68 MiB
    assets by chunk 2.76 MiB (id hint: vendors)
      assets by status 1.46 MiB [big]
        asset vendors-node_modules_graphiql_esm_index_js-node_modules_isomorphic-fetch_fetch-npm-browserify-359772.app-en.min.js 816 KiB [compared for emit] [minimized] [big] (id hint: vendors)
        asset vendors-node_modules_html2canvas_dist_html2canvas_js-node_modules_jspdf-autotable_dist_jspdf_-247707.app-en.min.js 682 KiB [compared for emit] [minimized] [big] (id hint: vendors) 1 related asset
      17 assets
  assets by status 3.08 MiB [emitted]
    asset ../bundle-stats.html 1.63 MiB [emitted] [compared for emit] [dev]
    asset ../bundle-stats.json 1.45 MiB [emitted] [compared for emit] [dev]
  orphan modules 2.45 MiB [orphan] 975 modules
  runtime modules 6.5 KiB 12 modules
  modules by path ./ 9.21 MiB
    modules by path ./node_modules/ 6.58 MiB 1155 modules
    modules by path ./src/ 2.64 MiB 352 modules
  ../data/faq.csv 12.5 KiB [built] [code generated]
  assert (ignored) 15 bytes [built] [code generated]

  WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
  This can impact web performance.
  Assets:
    vendors-node_modules_graphiql_esm_index_js-node_modules_isomorphic-fetch_fetch-npm-browserify-359772.app-en.min.js (816 KiB)
    vendors-node_modules_html2canvas_dist_html2canvas_js-node_modules_jspdf-autotable_dist_jspdf_-247707.app-en.min.js (682 KiB)

  en (webpack 5.9.0) compiled with 1 warning in 77746 ms
```

**Comment by ktw1016** [comment]
Hi @vio , indeed it's fixed.

I'll let you know if I find anything else. Thanks for the fix!
